### PR TITLE
Migrate qiskit.aqua.operators to qiskit.opflow

### PR DESCRIPTION
**Pull request by manoelmarques**
### Summary

Attention:

The files were migrated from qiskit-aqua with their commit history included. I understand that the merge should be a merge commit and not a squash and merge in order to preserve the previous Aqua history.

### Details and comments

Migrate:

qiskit/aqua/operators

from qiskit-aqua to qiskit-terra.
This commit migrates the above from the qiskit-aqua package to live in qiskit.opflow.
They were originally copied from:

https://github.com/Qiskit/qiskit-aqua/tree/049b9459474ecb36ceefcc8dbceb646d7f037e0f/qiskit/aqua/operators.

You can refer to that for the full development history leading up until this point.

Co-authored-by: Bryce Fuller <bgf024@gmail.com>
Co-authored-by: Christopher Wood <cjwood@us.ibm.com>
Co-authored-by: Donny Greenberg <dongreenberg2@gmail.com>
Co-authored-by: georgios-ts <45130028+georgios-ts@users.noreply.github.com>
Co-authored-by: Gonzalo <gonza.ar.cr@gmail.com>
Co-authored-by: Hayk Sargsyan <52532457+hay-k@users.noreply.github.com>
Co-authored-by: hhorii <hhorii@users.noreply.github.com>
Co-authored-by: Ikko Hamamura <ikkoham@users.noreply.github.com>
Co-authored-by: John Lapeyre <jlapeyre@users.noreply.github.com>
Co-authored-by: Julien Gacon <jul@zurich.ibm.com>
Co-authored-by: Manoel Marques <Manoel.Marques@ibm.com>
Co-authored-by: Matthew Treinish <mtreinish@kortar.org>
Co-authored-by: Max Rossmannek <oss@zurich.ibm.com>
Co-authored-by: Miroslav Tomasik <mirotomasik@gmail.com>
Co-authored-by: Pauline Ollitrault <49036540+paulineollitrault@users.noreply.github.com>
Co-authored-by: Richard Chen <chenrich@us.ibm.com>
Co-authored-by: Rochisha Agarwal <45593458+rochisha0@users.noreply.github.com>
Co-authored-by: Romain Fouilland <32517197+romainfd@users.noreply.github.com>
Co-authored-by: Stephen Wood <40241007+woodsp-ibm@users.noreply.github.com>
Co-authored-by: Takashi Imamichi <31178928+t-imamichi@users.noreply.github.com>
Co-authored-by: Wen Guan <wen.guan@cern.ch>
Co-authored-by: Zoufalc <40824883+Zoufalc@users.noreply.github.com>

### CODE REVIEW

File: setup.py
Comment (mtreinish) [inline setup.py:39]: I thought we were making scikit learn an optional dependency?

Comment (manoelmarques) [reply setup.py]: Changed it to optional

**Comment by mtreinish** [comment]
Adding on hold label to block mergify from accidentally squash and merging this since it requires a merge commit.

**Comment by t-imamichi** [comment]
#5261 is likely to conflict with this PR because opflow depends on the conventional Pauli operator and #5261 introduces new Pauli operator. We need to consider which to merge first.

**Review by mtreinish — APPROVED** [review]
LGTM as a code import. Some thing will need to be cleaned up post merging though. Primarily the docs aren't being included in the builds. Additionally, tests are being added in a follow on import commit.. Also, we discussed the legacy modules which are still required but should be removed prior to release (they're just there for ease of migration).

**Comment by mtreinish** [comment]
@t-imamichi I'm going to merge this first because it requires so manual intervention to preserve the commit history. We can rebase #5261 after this merges, there are already a bunch of follow up pieces that will be needed to do after this. It might have to wait until the opflow tests get migrated in a later commit (it wasn't a clean merge because of dependency on the algorithms module)

**Pull request merged**
